cmd/sonarExecuteScan.go

```diff
@@ -82,8 +85,25 @@ func sonarExecuteScan(config sonarExecuteScanOptions, _ *telemetry.CustomData, i
 	downloadClient.SetOptions(piperhttp.ClientOptions{TransportTimeout: 20 * time.Second})
 	// client for talking to the SonarQube API
 	apiClient := &piperhttp.Client{}
-	//TODO: implement certificate handling
-	apiClient.SetOptions(piperhttp.ClientOptions{TransportSkipVerification: true})
+	proxy := config.Proxy
+	options := piperhttp.ClientOptions{}
+	if proxy != "" {
+		transportProxy, err := url.Parse(proxy)
+		host, port, _ := net.SplitHostPort(transportProxy.Host)
+		javaToolOptions := fmt.Sprintf("-Dhttp.proxyHost=%v -Dhttp.proxyPort=%v", host, port)
+		os.Setenv("JAVA_TOOL_OPTIONS", javaToolOptions)
+		if err != nil {
+			log.Entry().WithError(err).Fatalf("Failed to parse proxy string %v into a URL structure", proxy)
+		}
+
+		options = piperhttp.ClientOptions{TransportProxy: transportProxy}
```

Comment:
I think we don't need to recreate options
```suggestion
		options.TransportProxy = transportProxy
```